index/scorch/segment/zap/dict.go

```diff
@@ -191,13 +192,55 @@ func (d *Dictionary) FuzzyIterator(term string,
 	return rv
 }
 
+func (d *Dictionary) OnlyIterator(onlyTerms [][]byte,
+	includeCount bool) segment.DictionaryIterator {
+
+	rv := &DictionaryIterator{
+		d:         d,
+		omitCount: !includeCount,
+	}
+
+	var buf bytes.Buffer
+	builder, err := vellum.New(&buf, nil)
+	if err != nil {
+		rv.err = err
+		return rv
+	}
+	for _, term := range onlyTerms {
+		err = builder.Insert(term, 0)
+		if err != nil {
+			rv.err = err
+			return rv
+		}
+	}
+	err = builder.Close()
+	if err != nil {
+		rv.err = err
+		return rv
+	}
+
+	onlyFST, err := vellum.Load(buf.Bytes())
+	if err != nil {
+		rv.err = err
+		return rv
+	}
+
+	itr, err := d.fst.Search(onlyFST, nil, nil)
+	if err == nil {
+		rv.itr = itr
```

Comment:
looks like you have an rv.err = err pattern that you're following, except for the fst.Search() call?

Reply:
good point, looks like that same bug applies to the regexp one above i copied from

Reply:
seems like that bug is in ALL of the dictionary iterators

Reply:
oh i see, presumably it was because they didn't want vellum.ErrIteratorDone to be treated as error further up the stack.  but we should explicitly check that...

Reply:
pushed another commit to try and at least make it more consistent